Samples/CameraCaptureUI/cpp-winui/cpp-winui/MainPage.xaml.cpp

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "MainPage.xaml.h"
+#if __has_include("MainPage.g.cpp")
+#include "MainPage.g.cpp"
+#endif
+
+namespace winrt
+{
+    using namespace Microsoft::UI::Xaml;
+    using namespace Microsoft::UI::Xaml::Controls;
+    using namespace Microsoft::UI::Xaml::Media;
+    using namespace Microsoft::UI::Xaml::Media::Animation;
+    using namespace Microsoft::UI::Xaml::Navigation;
+    using namespace Windows::UI::Xaml::Interop;
+}
+
+namespace winrt::cpp_winui::implementation
+{
+    cpp_winui::MainPage MainPage::current{ nullptr };
+
+    MainPage::MainPage()
+    {
+        InitializeComponent();
+        MainPage::current = *this;
+    }
+
+    void MainPage::NotifyUser(hstring const& strMessage, InfoBarSeverity const& severity)
+    {
+        // If called from the UI thread, then update immediately.
+        // Otherwise, schedule a task on the UI thread to perform the update.
+        if (this->DispatcherQueue().HasThreadAccess())
+        {
+            UpdateStatus(strMessage, severity);
+        }
+        else
+        {
+            DispatcherQueue().TryEnqueue([strongThis = get_strong(), this, strMessage, severity]
+                { UpdateStatus(strMessage, severity); });
```

Comment:
Here the references of `strMessage` & `severity` is getting passed, It won't ensure that at the time UpdateStatus gets executed in UI Thread these memory will still hold the expected values as the notification will be over by then and the memory can get released.

Reply:
This is the code that came as part of the sample template I used to create the sample.
I can see that other samples created using the template are using this implementation without any modifications.